dash/App/Accounts/Add/Components/index.js

```diff
@@ -0,0 +1,278 @@
+import React from 'react'
+import { useEffect, useRef, useState, useCallback } from 'react'
+
+import RingIcon from '../../../../../resources/Components/RingIcon'
+import { debounce } from '../../../../../resources/utils'
+import link from '../../../../../resources/link'
+import zxcvbn from 'zxcvbn'
+
+const navForward = async (newAccountType, accountData) =>
+  link.send('nav:forward', 'dash', {
+    view: 'accounts',
+    data: {
+      showAddAccounts: true,
+      newAccountType,
+      accountData
+    }
+  })
+
+const removeLineBreaks = (str) => str.replace(/(\r\n|\n|\r)/gm, '')
+
+function AddHotAccountWrapper({ children, title, svgName, summary, intro }) {
+  return (
+    <div className={'addAccountItem addAccountItemSmart addAccountItemAdding'}>
+      <div className='addAccountItemBar addAccountItemHot' />
+      <div className='addAccountItemWrap'>
+        <div className='addAccountItemTop'>
+          <div className='addAccountItemTopType'>
+            <div className='addAccountItemIcon'>
+              <div className='addAccountItemIconType addAccountItemIconHot'>
+                <RingIcon {...{ svgName }} />
+              </div>
+              <div className='addAccountItemIconHex addAccountItemIconHexHot' />
+            </div>
+            <div className='addAccountItemTopTitle'>{title}</div>
+          </div>
+          <div className='addAccountItemSummary'>{summary}</div>
+        </div>
+        <div className='addAccountItemOption'>
+          <div
+            className='addAccountItemOptionIntro'
+            onMouseDown={() => {
+              this.adding()
+              setTimeout(
+                () =>
+                  link.send('tray:action', 'navDash', {
+                    view: 'notify',
+                    data: { notify: 'hotAccountWarning', notifyData: {} }
+                  }),
+                800
+              )
+            }}
+          >
+            {intro}
+          </div>
+          <div className='addAccountItemOptionSetupFrames'>{children}</div>
+        </div>
+        <div className='addAccountItemFooter' />
+      </div>
+    </div>
+  )
+}
+
+function EnterSecret({ newAccountType, isValidSecret, title }) {
+  const [secret, setSecret] = useState('')
+  const [error, setError] = useState(null)
+
+  const updateInput = (e) => {
+    const value = removeLineBreaks(e.target.value)
+    setSecret(value)
```

Comment:
do we want to remove line breaks here? just checking because entering this data is very sensitive and we want to be absolutely sure that the users PK or seed is exactly what they entered